src/main/java/org/broadinstitute/barclay/help/DocletUtils.java

```diff
@@ -1,46 +1,67 @@
 package org.broadinstitute.barclay.help;
 
-import com.sun.javadoc.FieldDoc;
-import com.sun.javadoc.PackageDoc;
-import com.sun.javadoc.ProgramElementDoc;
-import org.broadinstitute.barclay.utils.JVMUtils;
+import jdk.javadoc.doclet.DocletEnvironment;
 
-import java.lang.reflect.Field;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.PackageElement;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 /**
  * Package protected - Methods in the class must ONLY be used by doclets, since the com.sun.javadoc.* classes
  * are not available on all systems, and we don't want the GATK proper to depend on them.
  */
-class DocletUtils {
+public class DocletUtils {
+    final static Logger logger = LogManager.getLogger();
 
-    protected static Class<?> getClassForDoc(ProgramElementDoc doc) throws ClassNotFoundException {
-        return Class.forName(getClassName(doc, true));
+    public static Class<?> getClassForDeclaredElement(final Element docElement, final DocletEnvironment docEnv) {
+         return getClassForClassName(getClassName(docElement, docEnv));
     }
 
-    protected static Field getFieldForFieldDoc(FieldDoc fieldDoc) {
+    public static Class<?> getClassForClassName(final String className) {
         try {
-            Class<?> clazz = getClassForDoc(fieldDoc.containingClass());
-            return JVMUtils.findField(clazz, fieldDoc.name());
+            return Class.forName(className);
         } catch (ClassNotFoundException e) {
-            throw new RuntimeException(e);
+            // we got an Element for a class we can't find.  Maybe in a library or something
+            return null;
+        } catch (NoClassDefFoundError e) {
+            return null;
         }
     }
 
     /**
      * Reconstitute the class name from the given class JavaDoc object.
      *
-     * @param doc the Javadoc model for the given class.
-     * @return The (string) class name of the given class.
+     * @param element the Element for a class.
+     * @return The (string) class name of the given class. Maybe null if no class can be found.
+     */
+    protected static String getClassName(final Element element, final DocletEnvironment docEnv) {
```

Comment:
Actually, I did this wen I first started the port, but I think it may be possible to do this with type mirrors. I'll try it.

Reply:
It looks like there is a way to get a qualified name as you would reference it in source, but its not the same as the actual class name for nested classes/enums, i.e., I need:

`org.broadinstitute.barclay.argparser.CommandLinePluginUnitTest$TestPluginBase`

but it returns:

`org.broadinstitute.barclay.argparser.CommandLinePluginUnitTest.TestPluginBase`

There must be some way to do it but I don't see it, so I'm leaving this since it's essentially the same as what we previously used with the old doclet apis.